src/datasets/utils/deprecation_utils.py

```diff
@@ -1,13 +1,25 @@
+import logging
 import warnings
 from typing import Callable
 
 
+_emitted_deprecation_warnings = set()
+logger = logging.getLogger(__name__)
+
+
 def deprecated(deprecated_function: Callable):
```

Comment:
Could we allow a to add a message to help users remove the warning ?
Users must know that they can replace `dataset.rename_columns_` by `dataset = dataset.rename_column` for example